### PR TITLE
Add support for segments, convert shell scripts to golang

### PR DESCRIPTION
**Pull request by leizor**

### CODE REVIEW

File: apply_recommendations/cmd/apply/main.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+func main() {
+	workingDir := flag.String("working-dir", "./", "path to working directory")
+	dryRun := flag.Bool("dry-run", false, "dry run; run terraform plan instead of terraform apply")
```
Comment (replay) [inline apply_recommendations/cmd/apply/main.go:17]: personally, i would default this to `true`, but that might be personal preference.

File: apply_recommendations/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
 
-RUN apk add --no-cache jq
+COPY . /opt
+WORKDIR /opt
 
-COPY --chmod=755 apply_recommendations.sh /opt/apply_recommendations.sh
+RUN CGO_ENABLED=0 GOOS=linux go build -o /apply-recommendations cmd/apply/main.go
 
-ENTRYPOINT ["/opt/apply_recommendations.sh"]
+FROM gcr.io/distroless/base-debian11
```
Comment (replay) [inline apply_recommendations/Dockerfile:8]: sorry, I'm not familiar with what this image contains, isn't "distroless" and "debian" kind of a contradiction?

**Review by replay — APPROVED** [review]
LGTM, thanks!

File: apply_recommendations/Dockerfile (hunk 2 above)
Comment (leizor) [reply apply_recommendations/Dockerfile:8]: Good point 😆 

I pulled it from [here](https://github.com/GoogleContainerTools/distroless?tab=readme-ov-file#docker); it sounded like about the right image to use.

**Pull request merged**